lib/bolt/pal.rb

```diff
@@ -97,6 +97,8 @@ def in_bolt_compiler
           alias_types(compiler)
           begin
             yield compiler
+          rescue Bolt::Error => err
```

Comment:
How did this come up?

Reply:
Running tasks directly can result in a Bolt::Error, we should be throwing it. Previously this primarily handled plans and our external task validation made sure we wouldn't run into an error.